src/django_rich/test.py

```diff
@@ -157,12 +162,82 @@ class RichTestRunner(DiscoverRunner.test_runner):
     resultclass = RichTextTestResult
 
 
+class RichTestSuite(unittest.TestSuite):
+    def run(
+        self, result: unittest.TestResult, debug: bool = False
+    ) -> unittest.TestResult:
+        topLevel = False
+        if getattr(result, "_testRunEntered", False) is False:
+            result._testRunEntered = topLevel = True  # type: ignore [attr-defined]
+
+        for index, test in enumerate(self):
+            if result.shouldStop:
+                break
+
+            if _isnotsuite(test):
+                self._tearDownPreviousClass(test, result)  # type: ignore [attr-defined]
+                self._handleModuleFixture(test, result)  # type: ignore [attr-defined]
+                self._handleClassSetUp(test, result)  # type: ignore [attr-defined]
+                result._previousTestClass = test.__class__  # type: ignore
+                # [attr-defined]
+                if getattr(test.__class__, "_classSetupFailed", False) or getattr(
+                    result, "_moduleSetUpFailed", False
+                ):
+                    continue
+            if not debug:
+                start_time = time.perf_counter()
+                test(result)
+                if isinstance(result, RichTextTestResult) and isinstance(
+                    test, TestCase
+                ):
```

Comment:
why do you check `isinstance(test, TestCase)` here?

Reply:
also, why the check on result? I think it's safe to assume if our test suite class is in use, our result class is. anyone subclassing should ensure that holds.